testing/src/main/java/io/airlift/testing/Closeables.java

```diff
@@ -35,12 +36,25 @@ public static void closeQuietly(Closeable... closeables)
 
     public static void closeAll(Closeable... closeables)
             throws IOException
+    {
+        try {
+            closeAll((AutoCloseable[]) closeables);
+        }
+        catch (Exception e) {
+            propagateIfPossible(e, IOException.class);
+            // Not possible
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void closeAll(AutoCloseable... closeables)
```

Comment:
The usages of `IOException` need to be changed

Reply:
This is why i retained `closeAll(Closeable...)`

Reply:
BTW I'd rather remove `closeAll(Closeable...)` if we can afford breakage.
(there is no reasonable gradual deprecation path I can think of)

Reply:
In the code below, we shouldn't wrap the exception with an IOException for this method:

```java
 catch (Throwable e) {
    if (rootCause == null) {
        rootCause = new IOException(e);
    }
```

Reply:
good catch